File: Service/Task.Entities/TaskModel.cs
using System;
using System.Collections.Generic;

namespace Task.Entities
{
    public class TaskModel
    {

        private int taskId;

        public int TaskId
        {
            get { return taskId; }
            set { taskId = value; }
        }

        private ParentTaskModel parentTask;

        public ParentTaskModel ParentTask
        {
            get { return parentTask; }
            set { parentTask = value; }
        }

        private int? parentTaskId;

        public int? ParentTaskId
        {
            get { return parentTaskId; }
            set { parentTaskId = value; }
        }

        private string taskDescription;

        public string TaskDescription
        {
            get { return taskDescription; }
            set { taskDescription = value; }
        }


        private DateTime? startDate;

        public DateTime? StartDate
        {
            get { return startDate; }
            set { startDate = value; }
        }

        private DateTime? endDate;

        public DateTime? EndDate
        {
            get { return endDate; }
            set { endDate = value; }
        }

        private int priority;

        public int Priority
        {
            get { return priority; }
            set { priority = value; }
        }

        private bool isClosed;

        public bool IsClosed
        {
            get { return isClosed; }
            set { isClosed = value; }
        }

        //public ICollection<TaskModel> ToList()
        //{
        //    throw new NotImplementedException();
        //}
    }
}

File: PM/UI/project-management/src/app/model/task-model.ts
import { ParentTaskModel } from "./parent-task-model";
import { ProjectModel } from "./project-model";
import { UserModel } from "./user-model";


export interface TaskModel {
    taskId : number;
    parentTask?     : ParentTaskModel;
    parentTaskId? : number ;
    taskDescription: string ;
    startDate?     : Date;
    endDate?     : Date;
    priority: number;
    isClosed     : boolean;
    projectId: number;
    project: ProjectModel;
    userId: number;
    user: UserModel;
    
}
